backend/src/server/config/rateLimiter.ts

```diff
@@ -52,6 +52,15 @@ export const inviteUserRateLimit: RateLimitOptions = {
   keyGenerator: (req) => req.realIp
 };
 
+export const mfaRateLimit: RateLimitOptions = {
+  timeWindow: 60 * 1000,
+  max: 20,
+  keyGenerator: (req) => {
+    // fallback to IP to avoid global rate limiting when authorization is set to empty
+    return req.headers.authorization?.split(" ")[1] || req.realIp;
```

Comment:
ah hm.. will decode this and filter it based on the userId specifically

Reply:
I can't use the `req.mfa` here because for some reason, the rate limiter gets executed first